### DISCUSSION TITLE
Errors when email notifications is enabled without sendmail/smtp configured

### DISCUSSION THREAD

**Discussion by rfos** [directus/directus#12997] [Archive]
Hello,

I'm trying to export a collection and save it as .CSV in my library but it shutdown the application because I have no smtp/send mail configured on my environment. (Is it mandatory?)

I believe it happens because the application try to send a notification when the export is done. Would be possible set up the notification to be conditional it sendmail/smtp is configured or not? 

### To Reproduce

Choose export to Library instead of Download File

![2022-04-25 17_18_45- env - Untitled (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/217889/165120128-263a5320-9a6e-4813-aff3-ea6b5a750f30.png)

### Errors Shown

`17:07:19 Γ£¿ request completed POST 204 /utils/export/employees 30ms
17:07:20 ≡ƒÜ¿ Couldn't export employees: spawn /usr/sbin/sendmail ENOENT 
Error: spawn /usr/sbin/sendmail ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^

Error: spawn /usr/sbin/sendmail ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn /usr/sbin/sendmail',
  path: '/usr/sbin/sendmail',
  spawnargs: [ '-i', '-f', 'no-reply@directus.io', 'email@domain.com ]
}
The terminal process "C:\Windows\System32\cmd.exe /d /c npm run start" terminated with exit code: 1.`

<details>
<summary>More info</summary>

### What version of Directus are you using?

9.9.1

### What version of Node.js are you using?

16.14.2

### What database are you using?

MS SQL 

### What browser are you using?

Chrome

### How are you deploying Directus?

locally - Windows 11

</details>

**Comment by azrikahar** [comment]
> I'm trying to export a collection and save it as .CSV in my library but it shutdown the application because I have no smtp/send mail configured on my environment. (Is it mandatory?)

You should be able to resolve this by turning off email notification in your profile page:

![chrome_883PIkw1t9](https://user-images.githubusercontent.com/42867097/165123057-b3c16b11-88ce-4ae1-b546-c0fb538e4e2b.png)

> Would be possible set up the notification to be conditional it sendmail/smtp is configured or not?

This is a good point, but perhaps we can make sure the notification goes through (this will still show in the app), but not the email sending portion. More about this below.

---

I believe this delves more into something that was brought up before on what should happen during this "email notification enabled but the instance does not have email configured" scenario. Hence I moved this to discussion while we explore the potential solutions here.

Technically the same error would happen when mentioning a user in an item comments, but the error is catched so it does not crash the instance:

![chrome_XXuDDElHR1](https://user-images.githubusercontent.com/42867097/165122764-9f8d3926-fa1e-496a-9259-1f8afb2b717e.png)

**Reply by rfos** [reply]
``Maybe would be better to change the description of these functions. We are talking about notifications (second icon at the left bottom) or "receive an email when it's done? 

![2022-04-26 21_01_16-](https://user-images.githubusercontent.com/217889/165373284-a9f2ec0a-95c3-4d6a-8df4-ed1fed151dde.png)


In the meantime I try to configure the app to use my gmail.com account but there is something wrong here.

```

EMAIL_TRANSPORT="smtp"
## Email (SMTP Transport)
EMAIL_SMTP_POOL=true
EMAIL_SMTP_HOST="smtp.gmail.com"
EMAIL_SMTP_PORT=587
EMAIL_SMTP_SECURE=true # Use TLS
# EMAIL_SMTP_IGNORE_TLS=false
EMAIL_SMTP_USER="mymail@gmail.com"
EMAIL_SMTP_PASSWORD="myAppPassword"
```

The console log-returns that message:
```
21:01:32 Γ£¿ request completed PATCH 204 /users/me/track/page 196ms
21:01:32 ΓÜá∩╕Å  Email connection failed:
21:01:32 ΓÜá∩╕Å  15480:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:
```
